peps/pep-9999.rst

```diff
@@ -0,0 +1,616 @@
+PEP: 9999
+Title: Standardizing selection of Python package files across multiple sources
+Author: Michael Sarahan, msarahan@gmail.com
+Sponsor: Barry Warsaw, barry@python.org
+PEP-Delegate: <PEP delegate’s real name>
+Discussions-To: <REQUIRED: URL of current canonical discussion thread>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Requires: 777
+Created: 05-Nov-2024
+Post-History: <REQUIRED: dates, in dd-mmm-yyyy format, and corresponding links to PEP discussion threads>
+
+Abstract
+========
+
+Python installers are diverging with respect to how they handle multiple
+indexes. Pip merges all indexes before matching distributions, while uv
+matches distributions on one index before moving on to the next. Each
+approach has advantages and disadvantages. This PEP aims to standardize
+each of these behaviors, which are referred to as “version priority” and
+“index priority” respectively, so that community discussions and
+troubleshooting can share a common vocabulary. Codifying these behaviors
+will also facilitate building tools that rely on these behaviors.
+
+Motivation
+==========
+
+Goals
+-----
+
+- Establish standard for evaluation order of various levels of priority
+  between indexes and packages
+- Standardize verbiage of different approaches, so that cross-tool
+  discussions are talking “apples to apples”
+- Suggest standard configuration options, to minimize user cognitive
+  load between tools
+- Provide guidelines for how ecosystem tools should implement index
+  priority if they would like, and suggest reasons why they might want
+  to do so
+
+Non-goals
+---------
+
+- Do not relax the simplifying assumption that for a given distribution
+  filename, all indexes must have the same file. This has been expressed
+  as “all indexes are equal priority”, but can be more accurately
+  described as “`pip considers all distributions with the same name and
+  version to be interchangeable regardless of which index they came
+  from <https://github.com/astral-sh/uv/issues/171#issuecomment-1952079681>`__.”
+  This is a key assumption for how pip behaves today, and changing this
+  assumption is deemed too disruptive for this PEP.
+- Do not promote index priority as a solution to dependency confusion.
+  It is related, but `PEP 708 <https://peps.python.org/pep-0708/>`__ is
+  a better, more specific solution that does not require user
+  configuration
+- Do not dictate that every tool must implement every strategy
+
+Rationale
+=========
+
+This PEP proposes standards of installer behavior when using multiple
```

Comment:
As @ethanhs-nv points out, an important rationale is to align on terminology, and I think "version priority" and "index priority" are good terms for the algorithms (which is one reason to get the word "priority" in the title).